src/AppBundle/Resources/public/js/ui.deckimport.js

```diff
@@ -2,27 +2,32 @@
 
 (function ui_deckimport(ui, $)
 {
-
-    var name_regexp;
-
     ui.on_content_change = function on_content_change(event)
     {
         var text = $(content).val(),
                 slots = {},
                 faction_code,
                 faction_name;
 
-        text.match(name_regexp).forEach(function (token)
+        text.match(/[^\r\n]+/g).forEach(function (token)
         {
-            var qty = 1, name = token.trim(), card;
-            if(token[0] === '(') {
-                return;
-            }
-            if(name.match(/^(\d+)x (.*)/)) {
+            var qty = 1, name = token.trim(), card, packName;
+            if (name.match(/^(\d+)x? ([^(]+) \(([^)]+)\)/)) {
+              qty = parseInt(RegExp.$1, 10);
```

Comment:
The rest of this file is using 4 space indent, while a lot of the new `if`s are using 2 space. However, this may be worth addressing separately by introducing linting rules.

Reply:
will fix. agreed, the code base definitely needs coding standards enforced via phpcs and js/eslint, etc. let's deal with that separately.

Reply:
handled.